File: 1/sol2.py
def first_repeat():
    seen = set()
    acc = 0
    seen.add(acc)
    with open('input') as f:
        numbers = list(map(int, f))

    while True:
        for n in numbers:
            acc += n
            print(acc)
            if acc in seen:
                return acc
            seen.add(acc)


print('first is', first_repeat())

File: 1/sol2.php
<?php
$file = fopen("input", "r");
$filesize = filesize("input");
$lines = explode("\n", fread($file, $filesize));
fclose($file);

$acc = 0;
$seen = array();
$seen[$acc] = true;
while(true) {
	foreach($lines as $num) {
		if(empty($num))
			continue;
		$acc += $num;
		if(array_key_exists($acc, $seen)){
			die("$acc is repeated\n");
		}
		$seen[$acc] = true;
	}
}
?>

